src/main/kotlin/lotto/domain/LottoNumber.kt

```diff
@@ -1,7 +1,26 @@
 package lotto.domain
 
-data class LottoNumber(val value: Int) {
+class LottoNumber private constructor(private val value: Int) : Comparable<LottoNumber> {
+
     override fun toString(): String {
         return value.toString()
     }
+
+    companion object {
+        const val MIN_LOTTO_NUMBER = 1
+        const val MAX_LOTTO_NUMBER = 45
+        private val NUMBERS: Map<Int, LottoNumber> = (MIN_LOTTO_NUMBER..MAX_LOTTO_NUMBER).associateWith(::LottoNumber)
+
+        fun from(value: Int): LottoNumber {
+            return NUMBERS[value] ?: throw IllegalArgumentException("로또의 번호는 1~45번 까지 있습니다.")
+        }
+    }
+
+    override fun compareTo(other: LottoNumber): Int {
+        return when {
+            value > other.value -> 1
+            value == other.value -> 0
+            else -> -1
+        }
```

Comment:
아래의 코드로 개선해 보면 어떨까요?

```suggestion
        return value.compareTo(other.value)
```